data/modules/CargoRun/CargoRun.lua

```diff
@@ -0,0 +1,1148 @@
+					attacker:AIKill(Game.player)
+				end
+			end
+		end
+	elseif isPirateShip(ship) then
+		for i = 1, #pirate_ship do
+			if pirate_ship[i] == ship then
+				table.remove(pirate_ship, i)
+				if isEscortShip(attacker) then
+					Comms.ImportantMessage(l.TARGET_DESTROYED, attacker.label)
+					if #pirate_ship ~= 0 then
+						attacker:AIKill(pirate_ship[1])
+						escort_switch_target = Game.time + Engine.rand:Integer(90, 120)
+					end
+				end
+			end
+		end
+	end
+end
+
+local onShipFiring = function (ship)
+	if ship:IsPlayer() then return end
+
+	if isEscortShip(ship) and Game.time >= escort_chatter_time then -- don't flood the control panel with messages
+		Comms.ImportantMessage(l.GUNS_GUNS_GUNS, ship.label)
+			escort_chatter_time = Game.time + Engine.rand:Integer(15, 45)
+	end
+end
+
+local onShipHit = function (ship, attacker)
+	if attacker == nil then return end
+	if not attacker:isa('Ship') then return end
+
+	if ship:IsPlayer() and #escort_ship ~= 0 and not isEscortShip(attacker) then
+		if Game.time >= escort_switch_target then -- don't switch between targets too quick
+			for i = 1, #escort_ship do
+				escort_ship[i]:AIKill(attacker) -- all escort ships respond
+			end
+			Comms.ImportantMessage(l.I_AM_ON_MY_WAY, escort_ship[1].label)
+			escort_switch_target = Game.time + Engine.rand:Integer(90, 120)
+		end
+
+	elseif isEscortShip(ship) then
+		if attacker:IsPlayer() then
+			Comms.ImportantMessage(l.CEASE_FIRE, ship.label)
+		elseif isPirateShip(attacker) then
+			if Game.time >= escort_switch_target then
+				ship:AIKill(attacker)
+				escort_switch_target = Game.time + Engine.rand:Integer(90, 120)
+			end
+			if Game.time >= escort_chatter_time then
+				Comms.ImportantMessage(l.I_NEED_HELP, ship.label)
+				escort_chatter_time = Game.time + Engine.rand:Integer(15, 45)
+			end
+		end
+
+	elseif isPirateShip(ship) then
+		if attacker:IsPlayer() then
+			if Game.time >= pirate_gripes_time then -- don't flood the control panel with messages
+				Comms.ImportantMessage(l["PIRATE_GRIPES_" .. Engine.rand:Integer(1, getNumberOf("PIRATE_GRIPES"))], ship.label)
+				pirate_gripes_time = Game.time + Engine.rand:Integer(30, 90)
+			end
+		elseif isEscortShip(attacker) then
+			if Game.time >= pirate_switch_target then
+				ship:AIKill(attacker)
+				pirate_switch_target = Game.time + Engine.rand:Integer(90, 120)
+			end
+		end
+	end
+end
+
+local onLeaveSystem = function (ship)
+	if ship:IsPlayer() then
+		nearbysystems = nil
+		pirate_ship = {}
+		escort_ship = {}
+	end
+end
+
+local unloadCargo = function (mission)
+	local cargotype = custom_cargo[mission.branch][mission.cargotype]
+	local uc = Game.player:RemoveEquip(cargotype, mission.cargo, "cargo")
+	if uc ~= mission.cargo then -- this is ugly
+		local list = Game.player:GetEquip("cargo")
+		for _, i in pairs(list) do
+			if i.l10n_key == cargotype.l10n_key then
+				Game.player:RemoveEquip(i, 1, "cargo")
+				uc = uc + 1
+				if uc == mission.cargo then break end
+			end
```

Comment:
Why do you need to do this? Does it have something to do with the case of the player jettisoning the cargo? Or is `Game.player:RemoveEquip` not behaving the way it should?